src/Symfony/Component/Messenger/DependencyInjection/MessengerPass.php

```diff
@@ -67,12 +67,12 @@ private function registerHandlers(ContainerBuilder $container)
 
         foreach ($container->findTaggedServiceIds($this->handlerTag, true) as $serviceId => $tags) {
             foreach ($tags as $tag) {
-                $handles = $tag['handles'] ?? $this->guessHandledClass($container, $serviceId);
+                $handles = $tag['handles'] ?? $this->guessHandledClass($r = $container->getReflectionClass($container->getDefinition($serviceId)->getClass()), $serviceId);
```

Comment:
The class in the definition can be a parameter like `%some_class_name%`, you should resolve it here (via `$container->getParameterBag()->resolveValue()`).

Reply:
fixed